component/azstorage/block_blob.go

```diff
@@ -71,6 +71,10 @@ type BlockBlob struct {
 // Verify that BlockBlob implements AzConnection interface
 var _ AzConnection = &BlockBlob{}
 
+const (
+	MaxBlocks = azblob.BlockBlobMaxStageBlockBytes * azblob.BlockBlobMaxBlocks
```

Comment:
While reading config if user has given config for block size, we shall check there if it exceeds the max allowed block size or not. Also, we can add the same test caes to our UT as well.